### PR TITLE
STM32: Add new STM board: Thunderpack v1.2

### PR DESCRIPTION
**Pull request by jgillick**
This PR makes the following STM port changes:

* Adds the [ThunderPack v1.2 board](https://github.com/jgillick/ThunderPack)
* Add support for SPI DotStar RGB status LED (APA102) to `Pin.c`.
* New linker script (see details below)
* Fixes a small issue with writing to NVM.

The new version of the ThunderPack has a 16MB SPI flash chip and an RGB status LED, so I couldn't just update the existing ThunderPack board files (they wouldn't be backward compatible)

## New Linker Script
The existing linker script (`STM32F411_nvm.ld`) didn't leave enough firmware space when compiling support for an external flash chip. The new linker script continues to support on-device NVM and shrinks the file system (it's stored on the flash chip anyway) to make room for the larger firmware.

## New Board

compact
original_and_compact

### CODE REVIEW

**Comment by hierophect** [comment]
Since you've included a flash chip, you could consider removing the filesystem entirely and getting yourself some additional flash. We only keep that option for chips that may disable it for some reason or other, and a 32kb filesystem has pretty limited utility anyway.

File: ports/stm/boards/STM32F411_nvm_flash.ld [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    more space for the CircuitPython firmware and less for the filesystem
+    (since the filesystem will be on the external flash chip).
+*/
+
+/* Specify the memory areas */
+MEMORY
+{
+    FLASH (rx)          : ORIGIN = 0x08000000, LENGTH = 512K /* entire flash */
+    FLASH_ISR (rx)      : ORIGIN = 0x08000000, LENGTH = 16K /* sector 0 */
+    FLASH_FS (rx)       : ORIGIN = 0x08004000, LENGTH = 32K /* sectors 1,2 are 16K */
```
Comment (hierophect) [inline ports/stm/boards/STM32F411_nvm_flash.ld:12]: You can just remove this, since you default to an external flash.

**Review by hierophect — CHANGES_REQUESTED** [review]
Looks good apart from your mpconfigboard.mk conflict.

File: ports/stm/boards/STM32F411_nvm_flash.ld (hunk 1 above)
Comment (jgillick) [reply ports/stm/boards/STM32F411_nvm_flash.ld]: Awesome! I wasn't sure that would be possible...but it also seemed like a waste to reserve space that wouldn't be used.

Comment (hierophect) [reply ports/stm/boards/STM32F411_nvm_flash.ld]: It still seems like a shame to me that NVM has to take up a full sector, but it just hasn't been high priority enough to spend the time to fix, especially when external flash makes it irrelevant anyway.

**Review by hierophect — APPROVED** [review]
LGTM! I'll have to make a note to pick one of these up for myself when I have a second.

**Comment by jgillick** [comment]
Thanks @hierophect. I'll be ordering a production run soon. If you want one, let me know and I'll ship one off to you.

**Pull request merged**